server/src/modules/auth/auth.service.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  BadRequestException,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { AuthDto } from './dto/auth.dto';
+import { CreateUserDto } from '../users/dto';
+import { UsersService } from '../users/users.service';
+import { RefreshTokensService } from '../refresh-token/refresh-tokens.service';
+import * as bcrypt from 'bcrypt';
+import { authServiceErrorMessages } from './auth.constants';
+import { AuthTokens, JwtPayload } from './types';
+import { JwtService } from '@nestjs/jwt';
+import { AppConfigService } from 'src/config/app-config.service';
+import { UserEntity } from '../users/user.entity';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private readonly usersService: UsersService,
+    private readonly refreshTokensService: RefreshTokensService,
+    private readonly jwtService: JwtService,
+    private readonly appConfigService: AppConfigService,
+  ) {}
+
+  async signUp(createUserDto: CreateUserDto): Promise<AuthTokens> {
+    const userExists = await this.usersService
+      .findOne({ email: createUserDto.email })
+      .catch(() => null);
+
+    if (userExists) {
+      throw new BadRequestException(
```

Comment:
For already existing entity it's better to use ConlictException (409)
```suggestion
      throw new ConflictException(
```